src/simplewallet/simplewallet.cpp

```diff
@@ -1744,43 +1833,32 @@ bool simple_wallet::print_ring(const std::vector<std::string> &args)
     PRINT_USAGE(USAGE_PRINT_RING);
     return true;
   }
+  std::string key_image_or_tx_id = args[0];
 
-  if (!epee::string_tools::hex_to_pod(args[0], key_image))
-  {
-    fail_msg_writer() << tr("Invalid key image");
-    return true;
-  }
-  // this one will always work, they're all 32 byte hex
-  if (!epee::string_tools::hex_to_pod(args[0], txid))
+  if (!epee::string_tools::hex_to_pod(key_image_or_tx_id, key_image)
```

Comment:
Looks to me as if there is no need to convert the same string twice, and no need for two variables `key_image` and `txid`, just one dummy one because the result is not really used.